core/src/main/java/com/graphhopper/storage/index/QueryResult.java

```diff
@@ -49,8 +51,10 @@ public QueryResult(double queryLat, double queryLon) {
     }
 
     /**
-     * @return the closest matching node. -1 if nothing found, this should be avoided via a call of
-     * 'isValid'
+     * Returns the closest matching node. This is either a tower node of the base graph
+     * or a virtual node (see also {@link com.graphhopper.routing.QueryGraph#lookup(List)}).
```

Comment:
I agree that we don't have a mention every time that virtual nodes might be involved such as in the routing algorithms. But I think in this case it's important to explain the connection to QueryGraph.lookup since this changes the behavior of this method (a virtual node may then be returned).